Description: Use the system botan library instead of the bundled one.
Author: Felix Geyer <debfx@ubuntu.com>
Last-Reviewed: 2014-08-19

---
 src/libs/3rdparty/botan/botan.pri        |   59 +------------------------------
 src/libs/ssh/sshcryptofacility.cpp       |    5 ++
 src/libs/ssh/sshcryptofacility_p.h       |    2 +
 src/libs/ssh/sshkeyexchange.cpp          |    6 +++
 src/libs/ssh/sshkeygenerator.cpp         |    5 ++
 src/libs/ssh/sshkeypasswordretriever_p.h |    1 
 6 files changed, 21 insertions(+), 57 deletions(-)

Index: qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshcryptofacility.cpp
===================================================================
--- qtcreator-3.4.0+dfsg~b1.orig/src/libs/ssh/sshcryptofacility.cpp	2015-03-07 17:26:46.077464409 -0600
+++ qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshcryptofacility.cpp	2015-03-07 17:26:46.073464409 -0600
@@ -38,6 +38,11 @@
 #include "sshpacket_p.h"
 
 #include <botan/botan.h>
+#include <botan/cbc.h>
+#include <botan/dsa.h>
+#include <botan/pkcs8.h>
+#include <botan/pubkey.h>
+#include <botan/rsa.h>
 
 #include <QDebug>
 #include <QList>
Index: qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshcryptofacility_p.h
===================================================================
--- qtcreator-3.4.0+dfsg~b1.orig/src/libs/ssh/sshcryptofacility_p.h	2015-03-07 17:26:46.077464409 -0600
+++ qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshcryptofacility_p.h	2015-03-07 17:26:46.073464409 -0600
@@ -32,9 +32,11 @@
 #define SSHCRYPTOFACILITY_P_H
 
 #include <botan/botan.h>
+#include <botan/hmac.h>
 
 #include <QByteArray>
 #include <QScopedPointer>
+#include <QSharedPointer>
 
 namespace QSsh {
 namespace Internal {
Index: qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeyexchange.cpp
===================================================================
--- qtcreator-3.4.0+dfsg~b1.orig/src/libs/ssh/sshkeyexchange.cpp	2015-03-07 17:26:46.077464409 -0600
+++ qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeyexchange.cpp	2015-03-07 17:26:46.073464409 -0600
@@ -37,6 +37,12 @@
 #include "sshincomingpacket_p.h"
 
 #include <botan/botan.h>
+#include <botan/dh.h>
+#include <botan/dl_group.h>
+#include <botan/dsa.h>
+#include <botan/pem.h>
+#include <botan/pubkey.h>
+#include <botan/rsa.h>
 
 #ifdef CREATOR_SSH_DEBUG
 #include <iostream>
Index: qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeygenerator.cpp
===================================================================
--- qtcreator-3.4.0+dfsg~b1.orig/src/libs/ssh/sshkeygenerator.cpp	2015-03-07 17:26:46.077464409 -0600
+++ qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeygenerator.cpp	2015-03-07 17:26:46.073464409 -0600
@@ -36,6 +36,11 @@
 #include "sshpacket_p.h"
 
 #include <botan/botan.h>
+#include <botan/der_enc.h>
+#include <botan/dsa.h>
+#include <botan/pem.h>
+#include <botan/pkcs8.h>
+#include <botan/rsa.h>
 
 #include <QDateTime>
 #include <QInputDialog>
Index: qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeypasswordretriever_p.h
===================================================================
--- qtcreator-3.4.0+dfsg~b1.orig/src/libs/ssh/sshkeypasswordretriever_p.h	2015-03-07 17:26:46.077464409 -0600
+++ qtcreator-3.4.0+dfsg~b1/src/libs/ssh/sshkeypasswordretriever_p.h	2015-03-07 17:26:46.073464409 -0600
@@ -31,6 +31,7 @@
 #define KEYPASSWORDRETRIEVER_H
 
 #include <botan/botan.h>
+#include <botan/ui.h>
 
 #include <string>
 
